libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway.hs

```diff
@@ -328,3 +356,181 @@ instance
+
+data ConwayCertExecContext era = ConwayCertExecContext
+  { ccecWithdrawals :: !(Map (Network, Credential 'Staking (EraCrypto era)) Coin)
+  , ccecVotes :: !(VotingProcedures era)
+  }
+  deriving (Generic, Eq, Show)
+
+instance Era era => Arbitrary (ConwayCertExecContext era) where
+  arbitrary =
+    ConwayCertExecContext
+      <$> arbitrary
+      <*> arbitrary
+
+instance
+  c ~ EraCrypto era =>
+  Inject
+    (ConwayCertExecContext era)
+    (Map (Network, Credential 'Staking c) Coin)
+  where
+  inject = ccecWithdrawals
+
+instance Inject (ConwayCertExecContext era) (VotingProcedures era) where
+  inject = ccecVotes
+
+instance Era era => ToExpr (ConwayCertExecContext era)
+
+instance
+  IsConwayUniv fn =>
+  ExecSpecRule fn "CERT" Conway
+  where
+  type ExecContext fn "CERT" Conway = ConwayCertExecContext Conway
+
+  environmentSpec _ = certEnvSpec
+  stateSpec _ _ = certStateSpec
+  signalSpec _ env st =
+    txCertSpec env st
+      <> constrained disableRegCerts
+      <> constrained disableDRepRegCerts
+    where
+      disableRegCerts :: Term fn (ConwayTxCert Conway) -> Pred fn
+      disableRegCerts cert =
+        (caseOn cert)
+          ( branch $ \delegCert ->
+              (caseOn delegCert)
+                (branch $ \_ _ -> False)
+                (branch $ \_ _ -> True)
+                (branch $ \_ _ -> True)
+                (branch $ \_ _ _ -> True)
+          )
+          (branch $ \_ -> True)
+          (branch $ \_ -> True)
+      -- ConwayRegDRep certificates seem to trigger some kind of a bug in the
+      -- MAlonzo code where it somehow reaches an uncovered case.
+      --
+      -- TODO investigate what's causing this bug and try to get rid of this
+      -- constraint
+      disableDRepRegCerts :: Term fn (ConwayTxCert Conway) -> Pred fn
+      disableDRepRegCerts cert =
+        (caseOn cert)
+          (branch $ \_ -> True)
+          (branch $ \_ -> True)
+          ( branch $ \govCert ->
+              (caseOn govCert)
+                (branch $ \_ _ _ -> False)
+                (branch $ \_ _ -> True)
+                (branch $ \_ _ -> False)
+                (branch $ \_ _ -> True)
+                (branch $ \_ _ -> True)
+          )
+
+  runAgdaRule env st sig =
+    first (\e -> OpaqueErrorString (T.unpack e) NE.:| [])
+      . computationResultToEither
+      $ Agda.certStep env st sig
+
+data ConwayRatifyExecContext era = ConwayRatifyExecContext
+  { crecTreasury :: Coin
+  , crecGovActionMap :: Map (GovActionId (EraCrypto era)) (GovActionState era)
+  }
+  deriving (Generic, Eq, Show)
+
+instance ToExpr (PParamsHKD StrictMaybe era) => ToExpr (ConwayRatifyExecContext era)
+
+instance
+  ( Era era
+  , Arbitrary (PParamsHKD StrictMaybe era)
+  ) =>
+  Arbitrary (ConwayRatifyExecContext era)
+  where
+  arbitrary =
+    ConwayRatifyExecContext
+      <$> arbitrary
+      <*> (Map.mapWithKey fixupGAIDs <$> arbitrary)
+    where
+      fixupGAIDs gaId gas = gas {gasId = gaId}
```

Comment:
I don't think it really matters here, since this is just a local function, but in general I think it's a good idea to not overuse `fixup` when naming functions